### PR TITLE
docs(rxSearchBox): FRMW-111 - Create rxSearchBox API Docs

### PR DESCRIPTION
**Pull request by bry4n**
JIRA: https://jira.rax.io/browse/FRMW-111

* Create JS Docs for component and directive
* Removed Options table 

### LGTMs

- [x] Dev LGTM
- [x] Dev LGTM

### Screenshot

rxsearchbox encoreui

### CODE REVIEW

File: src/rxSearchBox/rxSearchBox.js [hunk 1]
```diff
@@ -1,4 +1,34 @@
...
+ * @name rxSearchBox.directive:rxSearchBox
+ * @ngdoc directive
+ * @restrict E
+ * @description
+ * The rxSearchBox directive behaves similar to the HTML "Search" input type. When
+ * the search box is not empty, an "X" button within the element will allow you to
+ * clear the value. Once clear, the "X" will disappear. A disabled search box cannot
+ * be cleared of its value via the "X" button because the button will display.
+ *
+ * @param {String} ng-model - Model value to bind the search value. Required
```
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: Unless you are using the [square bracket notation](http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values) around the param name, it is assumed to be required. Please remove "Required" from the description.

File: src/rxSearchBox/rxSearchBox.js [hunk 2]
```diff
@@ -1,4 +1,34 @@
...
+ * @param {Boolean} ng-disabled - Boolean value to enable/disable the search box. Optional
```
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: Per jsdoc [@param](http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values) documentation, you can define this param as optional using the square bracket syntax around the param name as such:

```javascript
/**
 * @param {Boolean} [ng-disabled] Boolean value to enable/disable the search box.
 */
```

Additionally, ngdocs assumes that you want to start an unordered list when it sees a hyphen in the description. Please do not use hyphens in the description unless you actually want an unordered list.

File: src/rxSearchBox/rxSearchBox.js [hunk 3]
```diff
@@ -1,4 +1,34 @@
...
+ * @param {String} ng-placeholder - String to override the default placeholder (Search...). Optional
```
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: Please use [square bracket notation](http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values) with default value syntax to define the default value, as such:

```javascript
/**
 * @param {String} [ng-placeholder='Search...'] String to override the default placeholder.
 */
```

File: src/rxSearchBox/rxSearchBox.js [hunk 4]
```diff
@@ -1,4 +1,34 @@
...
+ *
+ * @example
+ * <pre>
+ * <rx-search-box class="search-box" ng-model="searchModel"></rx-search-box>
```
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: please remove the `class` attribute and its value from the example

File: src/rxSearchBox/rxSearchBox.js [hunk 5]
```diff
@@ -1,4 +1,34 @@
+/**
+ * @ngdoc overview
+ * @name rxSearchBox
+ * @description
+ * # rxSearchBox Component
+ *
+ * rxSearchBox component provides a directive to create search input box.
```
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: maybe...

```
The rxSearchBox component provides functionality around creating a search input box.
```

**Comment by CITguy** [comment]
Please add a link to the api documentation directly below the stability badge.

```
[API Documentation](ngdocs/index.html#/api/rxSearchBox)
```

File: src/rxSearchBox/rxSearchBox.js (hunk 2 above)
Comment (bry4n) [inline src/rxSearchBox/rxSearchBox.js]: Used square bracket syntax.

```js
/**
 * @param {String} [ng-model] Model value to bind the search value.
 * @param {Boolean} [ng-disabled] Boolean value to enable/disable the search box.
 * @param {String} [ng-placeholder='Search...'] String to override the default placeholder.
 */
```
and ngdoc generated this:

```html
<rx-search-box
       [ng-model]="{String}"
       [ng-disabled]="{Boolean}"
       ng-placeholder="{String}">
</rx-search-box>
```

Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: ugh, ngdocs screws up jsdoc once again... here's a cheat sheet

```javascript
/**
 * @param {String} foo Required Parameter
 * @param {String=} bar Optional Parameter (no default value)
 * @param {String=} [bang='biz'] Optional Parameter (with default value)
 */
```

Comment (bry4n) [inline src/rxSearchBox/rxSearchBox.js]: Updated

File: src/rxSearchBox/rxSearchBox.js [hunk 6]
```diff
@@ -1,4 +1,34 @@
...
+ * @ngdoc directive
+ * @restrict E
+ * @description
+ * The rxSearchBox directive behaves similar to the HTML "Search" input type. When
+ * the search box is not empty, an "X" button within the element will allow you to
+ * clear the value. Once clear, the "X" will disappear. A disabled search box cannot
+ * be cleared of its value via the "X" button because the button will display.
+ *
+ * @param {String} ng-model Model value to bind the search value.
+ * @param {Boolean=} [ng-disabled] Boolean value to enable/disable the search box.
```
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: Please do not use square brackets around `ng-disabled`. It's generating the following...

screen shot 2015-09-11 at 1 10 13 pm

Only use square brackets if you have a default value you want to set.

**Comment by CITguy** [comment]
@bry4n, please rebase. #1194 made some base changes to the `rxSearchBox.js`.

**Comment by CITguy** [comment]
LGTM, please rebase and squash

**Comment by 100stacks** [comment]
Hey looks like you might have an open comment in  src/rxSearchBox/rxSearchBox.js on Line 1. LGTM.

File: src/rxSearchBox/rxSearchBox.js [hunk 7]
```diff
@@ -1,20 +1,33 @@
-/**
- * @nagdoc overview
+
```
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: Missing open comment `/**`

**Comment by bry4n** [comment]
@100stacks @CITguy 

 I fixed the jsdocs. :+1:

**Pull request merged**